### ISSUE TITLE
Unable to make posts with attachments

### ISSUE THREAD

**Issue by WizardShotTheFood** [valknight/Cohost.py#27]
Starting a few days ago, my bot (pokemonoftheday) gives the following error when it gets run:

```
Traceback (most recent call last):
  File "/media/d4d6/countertenor/private/potd/potdcohost.py", line 107, in <module>
    newPost = project.post("Today's Pokémon is #" + str(pokeNum) + ": " + pokeName + ", the " + pokeGenus, blocks, tags=['pokemon', pokeName, 'automated posting', 'bot'])
  File "/media/d4d6/countertenor/python/lib/python3.9/site-packages/cohost/models/project.py", line 197, in post
    attachment.uploadIfNot(req['postId'], self)
  File "/media/d4d6/countertenor/python/lib/python3.9/site-packages/cohost/models/block.py", line 88, in uploadIfNot
    dospacecreds = fetch('post', endpoint, {
  File "/media/d4d6/countertenor/python/lib/python3.9/site-packages/cohost/network.py", line 58, in fetch
    raise Exception(res)
Exception: [{'type': 'Body', 'errors': {'issues': [{'code': 'invalid_type', 'expected': 'number', 'received': 'string', 'path': ['content_length'], 'message': 'Expected number, received string'}], 'name': 'ZodError'}}]
```

Based on the comments in https://cohost.org/ghobot/post/1361443-huh, this appears to be related to AttachmentBlock. Updating to 0.2.5 does not help.

**Comment by cohoghobo** [comment]
I can second this one, @WizardShotTheFood  actually linked my post above. Here's my traceback, it's pretty much the same 
```python
Traceback (most recent call last):
  File "/volume1/web_dev/morrowind_npcs/mwnpcs.py", line 179, in <module>
    main()
  File "/volume1/web_dev/morrowind_npcs/mwnpcs.py", line 163, in main
    p = project.post('',
  File "/volume1/web_dev/morrowind_npcs/mwnpcs_env/lib/python3.9/site-packages/cohost/models/project.py", line 197, in post
    attachment.uploadIfNot(req['postId'], self)
  File "/volume1/web_dev/morrowind_npcs/mwnpcs_env/lib/python3.9/site-packages/cohost/models/block.py", line 88, in uploadIfNot
    dospacecreds = fetch('post', endpoint, {
  File "/volume1/web_dev/morrowind_npcs/mwnpcs_env/lib/python3.9/site-packages/cohost/network.py", line 58, in fetch
    raise Exception(res)
Exception: [{'type': 'Body', 'errors': {'issues': [{'code': 'invalid_type', 'expected': 'number', 'received': 'string', 'path': ['content_length'], 'message': 'Expected number, received string'}], 'name': 'ZodError'}}]
``` 
As I mentioned in the comments to my post, I was able to get everything to work when I took the AttachmentBlock out of my list of blocks.

I found the same issue referenced in [a post from a few months ago](https://cohost.org/glotch/post/629076-acecourtbot-is-prett):

> also, when i first made a POST request using the cohost.py fetch method, i got an error back stating that the server "expected an int, not a string" when passing in an int parameter. the thing is, by all accounts, i was sending an int. after some poking and prodding, i discovered that the python requests library, and by extension the cohost library, sends its requests as url encoded by default, which didn't play nice with the POST request i guess. the cohost.py fetch method also doesn't let you set the content-type yourself, so i had to just use python's request library to send my own request with a json content type.

 and the original poster gave some [possible suggestions](https://cohost.org/ghobot/post/1361799-oh-hey-i-think-think#comment-2ac235ad-33f1-4575-abba-b3c7f9c6ce37): 

> luckily using the requests library to send a request with the json content-type rather than url encoded [is pretty straightforward from my experience](https://github.com/glotchalotch/acecourtbot-cohost/blob/master/main.py#L26) so hopefully you can get your bot running again soon! (assuming this is the solution)

My problems started on April 18, which based on the timing makes me think it's a problem with Digital Ocean that was mentioned in the [cohost patch notes yesterday](https://cohost.org/staff/post/1366571-cohost-patch-notes-vl)

**Comment by valknight** [comment]
I have a feeling this is tied to Cohost changing image uploads to be through their domain, and some potential changes that might have been made with that. I'll try and set aside some time to take a look at this 😄

**Comment by jkap** [comment]
`content_length` in the start request (https://github.com/valknight/Cohost.py/blob/a4aacd190ec87398eefc978a69e73462d38eae43/cohost/models/block.py#L88) needs to be a number, changing it to `postjson` is probably enough

**Comment by cohoghobo** [comment]
I just tried it out, and changing from `post` to `postjson` did the trick! Thanks for the help everyone

**Comment by valknight** [comment]
thanks @jkap ! i'll get a change to that effect merged in

**Comment by valknight** [comment]
Merged in! Thank you everyone in this thread - this fix is now implemented in 0.2.6
